### PR TITLE
RET-2180: Reply and Close Referral events

### PR DESCRIPTION
**Pull request by robin-ieong**
https://tools.hmcts.net/jira/browse/RET-2180

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: definitions/json/AuthorisationCaseField.json [hunk 1]
```diff
@@ -16829,6 +16829,546 @@
...
+    "CaseTypeId": "ET_Scotland",
+    "CaseFieldID": "hearingAndReferralDetails",
+    "UserRole": "caseworker-employment-legalrep-solicitor",
+    "CRUD": "CRU"
+  },
+  {
+    "CaseTypeId": "ET_Scotland",
+    "CaseFieldID": "hearingAndReferralDetails",
+    "UserRole": "caseworker-employment",
+    "CRUD": "CRU"
```
Comment (hsjhita) [inline definitions/json/AuthorisationCaseField.json:16842]: caseworker employment role should be R only

File: definitions/json/CaseEvent.json [hunk 2]
```diff
@@ -508,8 +508,23 @@
     "ShowEventNotes": "N",
     "ShowSummary": "Y",
     "CallBackURLAboutToStartEvent": "${ET_COS_URL}/replyReferral/aboutToStart",
-    "CallBackURLAboutToSubmitEvent": "",
-    "CallBackURLSubmittedEvent": "",
+    "CallBackURLAboutToSubmitEvent": "${ET_COS_URL}/replyReferral/aboutToSubmit",
```
Comment (hsjhita) [inline definitions/json/CaseEvent.json:511]: looking at EngWales, create referral missing submitted URL?

Comment (hsjhita) [reply definitions/json/CaseEvent.json:511]: is that EngWales specific?

File: definitions/json/ComplexTypes.json [hunk 3]
```diff
@@ -5730,6 +5730,77 @@
     "ElementLabel": "Short Description",
     "SecurityClassification": "Public"
   },
+  {
+    "ID": "replyDocumentCollection",
```
Comment (hsjhita) [inline definitions/json/ComplexTypes.json:5734]: Could this be replaced with DocumentUpload complextype?

**Review by JackReevies — APPROVED** [review]

**Review by JackReevies — APPROVED** [review]

**Review by hsjhita — APPROVED** [review]

**Pull request merged**